src/app/game.js

```diff
@@ -0,0 +1,222 @@
+import { log, arrayIsInArray, randomPosition } from '../utils/main'
+
+import { constants } from './config'
+
+export const generateGameData = (columns, rows) => {
+  // protect from invalid hacked configurations
+  if (
+    constants.GPGP_MARGIN > constants.MAX_MARGIN ||
+    constants.PROTECTED_ZONE_MARGIN > constants.MAX_MARGIN
+  ) {
+    log(
+      `error invalid configuration data detected on margin, one of the margins exceed max value of ${constants.MAX_MARGIN}`,
+      'error'
+    )
+    return { running: false }
+  }
+  const gpgpCenterPosition = randomPosition(columns, rows - 1)
+  const gpgpArea = calcArea(gpgpCenterPosition, constants.GPGP_MARGIN)
+  const pzArea = calcArea(gpgpCenterPosition, constants.PROTECTED_ZONE_MARGIN)
+  const gpgpAreaArray = getAreaArray(gpgpArea, columns, rows)
+  const pzAreaArray = getAreaArray(pzArea, columns, rows)
+
+  const bottlePosition = getEntityRandomPosition(columns, rows, pzAreaArray)
+  // push bottlePosition
+  const playerPosition = getEntityRandomPosition(
+    columns,
+    rows,
+    // push bottlePosition, player can't start in same position
+    [...pzAreaArray, bottlePosition]
+  )
+
+  // state.gameStatus
+  return {
+    running: true,
+    winner: undefined,
+    startingPosition: playerPosition,
+    playerStatus: {
+      position: playerPosition
+    },
+    bottleStatus: {
+      position: bottlePosition
+    },
+    turn: constants.PLASTIC_BOTTLE_ID,
+    directionStatus: constants.DIRECTIONS[0],
+    rounds: [],
+    rows: 0,
+    columns: 0,
+    gpgpCenterPosition,
+    // gpgpArea,
+    // pzArea,
+    gpgpAreaArray,
+    pzAreaArray
+  }
+}
+
+export const calcArea = (center, margin) => {
+  return {
+    topLft: [center[0] - margin, center[1] - margin],
+    topRgt: [center[0] + margin, center[1] - margin],
+    botRgt: [center[0] + margin, center[1] + margin],
+    botLft: [center[0] - margin, center[1] + margin]
+  }
+}
+
+export const getAreaArray = (area, columns, rows) => {
+  const areaArray = []
+  const { topLft, topRgt, botRgt, botLft } = area
+  const top = topLft[1]
+  const bot = botRgt[1] + 1
+  const lft = botLft[0]
+  const rgt = topRgt[0] + 1
+
+  for (let col = top; col < bot; col++) {
+    for (let row = lft; row < rgt; row++) {
+      let finalRow = row
+      let finalCol = col
+      // work with overflow negative
```

Comment:
you use this check at least twice in the app. It is better to extract it into a function or method

Reply:
I don't understand this recommendation
I only see it one time at src/app/game.js line 74, like you mention

can you point me the other place is is used

thanks @realiarthur

Reply:
I thought that lines #175 - #190 do the same check. Am I wrong?